src/Compilers/CSharp/Portable/Lowering/SyntheticBoundNodeFactory.cs

```diff
@@ -397,22 +397,52 @@ public MethodSymbol WellKnownMethod(WellKnownMember wm)
         /// <param name="sm">The desired special member</param>
         /// <returns>A symbol for the special member.</returns>
         public Symbol SpecialMember(SpecialMember sm)
+        {
+            return SpecialMember(sm, isOptional: false)!;
+        }
+
+        public Symbol? SpecialMember(SpecialMember sm, bool isOptional = false)
         {
             Symbol specialMember = Compilation.GetSpecialTypeMember(sm);
             if (specialMember is null)
             {
+                if (isOptional)
+                {
+                    return null;
+                }
+
                 RuntimeMembers.MemberDescriptor memberDescriptor = SpecialMembers.GetDescriptor(sm);
                 var diagnostic = new CSDiagnostic(new CSDiagnosticInfo(ErrorCode.ERR_MissingPredefinedMember, memberDescriptor.DeclaringTypeMetadataName, memberDescriptor.Name), Syntax.Location);
                 throw new MissingPredefinedMember(diagnostic);
             }
 
-            Binder.ReportUseSite(specialMember, Diagnostics, Syntax);
+            UseSiteInfo<AssemblySymbol> useSiteInfo = specialMember.GetUseSiteInfo();
+
+            if (isOptional)
+            {
+                if (useSiteInfo.DiagnosticInfo?.DefaultSeverity == DiagnosticSeverity.Error)
+                {
+                    return null;
+                }
+
+                // Not interested in warnings
+            }
+            else
+            {
+                Diagnostics.Add(useSiteInfo, Syntax);
+            }
+
             return specialMember;
         }
 
         public MethodSymbol SpecialMethod(SpecialMember sm)
         {
-            return (MethodSymbol)SpecialMember(sm);
+            return (MethodSymbol)SpecialMember(sm, isOptional: false)!;
```

Comment:
>! [](http://example.com/codeflow?start=69&length=1)

A nullable assertion would be preferred here since this method doesn't contain any null check. Nullable suppressions are used for cases where we have a null check but the compiler could not track.

Reply:
```suggestion
            var result = (MethodSymbol)SpecialMember(sm, isOptional: false);
            Debug.Assert(result is not null);
            return result;
```